lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,278 @@
+    #   Response if group doesn't exist:
+    #     {"code": 404, "message": "Not found"}
+    #   Response if groups are not supported:
+    #     {"code": 501, "error": "Groups are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_group(group)
+      raise "group must be provided" if group.blank?
+      parse get("/groups/#{URI.escape(group)}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain group '#{group}' from External IPAM."))
+    end
+
+    # Get a list of subnets for the given External IPAM group.
+    #
+    # Params:  1. group:  The name of the External IPAM group containing the subnet.
+    #
+    # Returns: An array of subnets on success, or a hash with an "error" key on failure.
+    #
+    # Responses from Proxy plugin:
+    #   Response if success: {"code": 200, "data": [
+    #     {subnet":"100.10.10.0","mask":"24","description":"Test Subnet 1"},
+    #     {subnet":"100.20.20.0","mask":"24","description":"Test Subnet 2"}
+    #   ]}
+    #   Response if no subnets exist in section.
+    #     {"code": 200, "data": []}
+    #   Response if section not found:
+    #     {"code": 404, "error": "Group not found in External IPAM"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM"}
+    def get_subnets_by_group(group)
+      raise "group must be provided" if group.blank?
+      unescaped_group = URI.unescape(group)
+      parse get("/groups/#{URI.escape(unescaped_group)}/subnets")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnets in group %s from External IPAM."), unescaped_group)
+    end
+
+    # Returns an array of subnets from External IPAM matching the given subnet.
+    #
+    # Params:  1. subnet:           The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24",
+    #                               IPv6 - "2001:db8:abcd:12::/124")
+    #          2. group(optional):  The name of the External IPAM group containing the subnet.
+    #
+    # Returns: A subnet on success, or a hash with an "error" key on failure.
+    #
+    # Responses from Proxy plugin:
+    #   Response if subnet(s) exists:
+    #     {"code": 200, "data": {"subnet": "44.44.44.0", "description": "", "mask":"29"}}
+    #   Response if subnet not exists:
+    #     {"code": 404, "error": "No subnets found"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_subnet(subnet, group = "")
+      raise "subnet cannot be nil" if subnet.nil?
+      parse get("/subnet/#{subnet}?group=#{URI.escape(group.to_s)}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnet %s from External IPAM."), subnet)
+    end
```

Comment:
Can you also add the raise when `error` is returned to align behaviour with  previous methods if possible?